test/test_vmap.py

```diff
@@ -640,35 +643,42 @@ def randn(size, device='cpu', dtype=torch.float):
     def randp1(size, device='cpu', dtype=torch.float):
         return torch.rand(size, device=device, dtype=dtype) + 1
 
+# Tests vmap(op, in_dims, out_dims)(*inputs) by comparing the output to a
+# (slow) sequential map+stack fallback.
+#
+# check_view: Test if the first returned output is a view of the first input
+# check_propagates_grad: Test if the operation propagates gradients.
+def _vmap_test(self, op, inputs, in_dims=0, out_dims=0,
+               check_view=False, check_propagates_grad=True):
+    result = vmap(op, in_dims, out_dims)(*inputs)
+    reference_result = reference_vmap(op, inputs, in_dims, out_dims)
+    self.assertEqual(result, reference_result)
+    op_has_single_return = not isinstance(result, tuple)
+
+    if check_view:
+        result_as_tuple = (result,) if op_has_single_return else result
+        for output in result_as_tuple:
+            self.assertEqual(output.data_ptr() - output.storage_offset() * output.element_size(),
```

Comment:
Note that you could try as well to check that `output._base is inputs[0] if inputs[0]._base is None else inputs[0]._base` ?

Reply:
I like that better, actually. I'll update the code to do that